DataFormats/NanoAOD/interface/FlatTable.h

```diff
@@ -104,40 +106,43 @@ namespace nanoaod {
     };
     RowView row(unsigned int row) const { return RowView(*this, row); }
 
-    template <typename T, typename C = std::vector<T>>
-    void addColumn(const std::string &name,
-                   const C &values,
-                   const std::string &docString,
-                   ColumnType type = defaultColumnType<T>(),
-                   int mantissaBits = -1) {
+    template <typename T, typename C>
+    void addColumn(const std::string &name, const C &values, const std::string &docString, int mantissaBits = -1) {
       if (columnIndex(name) != -1)
         throw cms::Exception("LogicError", "Duplicated column: " + name);
       if (values.size() != size())
         throw cms::Exception("LogicError", "Mismatched size for " + name);
-      check_type<T>(type);  // throws if type is wrong
-      auto &vec = bigVector<T>();
-      columns_.emplace_back(name, docString, type, vec.size());
-      vec.insert(vec.end(), values.begin(), values.end());
-      if (type == FloatColumn) {
-        flatTableHelper::MaybeMantissaReduce<T>(mantissaBits).bulk(columnData<T>(columns_.size() - 1));
+      if constexpr (std::is_same<T, bool>()) {
+        columns_.emplace_back(name, docString, ColumnType::Bool, uint8s_.size());
+        uint8s_.insert(uint8s_.end(), values.begin(), values.end());
+      } else if constexpr (std::is_same<T, float>()) {
+        columns_.emplace_back(name, docString, ColumnType::Float, floats_.size());
+        floats_.insert(floats_.end(), values.begin(), values.end());
+        flatTableHelper::MaybeMantissaReduce<float>(mantissaBits).bulk(columnData<float>(columns_.size() - 1));
+      } else {
+        ColumnType type = defaultColumnType<T>();
+        auto &vec = bigVector<T>();
+        columns_.emplace_back(name, docString, type, vec.size());
+        vec.insert(vec.end(), values.begin(), values.end());
```

Comment:
I'm missing why this if/else specialization is needed.
It looks like the code is already set up to be generic and on L126 we just need
```C++
flatTableHelper::MaybeMantissaReduce<type>(mantissaBits).bulk(columnData<type>(columns_.size() - 1));
```
(or is it "T")

Reply:
Indeed, you are right. I didn't look at `flatTableHelper::MaybeMantissaReduce` carefully enough.